pkg/gohai/platform/platform_darwin.go

```diff
@@ -27,13 +53,15 @@ func processIsTranslated() (bool, error) {
 	return false, err
 }
 
-func updateArchInfo(archInfo map[string]string, values []string) {
-	archInfo["kernel_name"] = values[0]
-	archInfo["hostname"] = values[1]
-	archInfo["kernel_release"] = values[2]
-	archInfo["machine"] = values[3]
-	archInfo["processor"] = strings.Trim(values[4], "\n")
-	archInfo["os"] = values[0]
+func updateArchInfo(archInfo map[string]string, uname *unix.Utsname) {
+	archInfo["kernel_name"] = utils.StringFromBytes(uname.Sysname[:])
```

Comment:
Unless I'm missing something, this could be replace by `string(uname.Sysname[:])`, no ?